File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/repository/transformer/RxGistTransformer.java
package ru.avystavkin.githubgists.repository.transformer;


import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import ru.avystavkin.githubgists.models.local.Gist;
import ru.avystavkin.githubgists.models.server.GistServer;

public class RxGistTransformer implements ObservableTransformer<GistServer, Gist> {
    @Override
    public ObservableSource<Gist> apply(Observable<GistServer> observable) {
        return observable
                .map(TransformerHelper::convertGistServer)
                .flatMap(Observable::just);
    }
}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/repository/github/transformer/RxGistTransformer.kt
package ru.avystavkin.githubgists.repository.github.transformer


import io.reactivex.Observable
import io.reactivex.ObservableSource
import io.reactivex.ObservableTransformer
import ru.avystavkin.githubgists.models.database.Gist
import ru.avystavkin.githubgists.models.server.GistServer

class RxGistTransformer : ObservableTransformer<GistServer, Gist> {
    override fun apply(observable: Observable<GistServer>): ObservableSource<Gist> {
        return observable
                .map { x -> TransformerHelper.convertGistServer(x) }
                .flatMap { x -> Observable.just(x) }
    }
}
